pre-commit

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+STAGED_GO_FILES=$(git diff --cached --name-only | grep ".go$" | grep -v "mock")
+
+for FILE in ${STAGED_GO_FILES}
+do
+    gofmt -w -s "${FILE}"
+    goimports -w "${FILE}"
+    git add "${FILE}"
+done
+
+if [[ -n "${STAGED_GO_FILES}" ]]; then
+    set -o errexit
+    go test --tags="unit integration" -race ./internal...
+    make fix-lint
+    set +o errexit
+    for FILE in ${STAGED_GO_FILES}
+    do
+        git add "${FILE}"
+    done
+
+    make gen-docs > /dev/null
+    git add docs
+fi
+
+STAGED_EVG_FILES=$(git diff --cached --name-only | grep "evergreen.yml$")
+
+for FILE in ${STAGED_EVG_FILES}
+do
+    evergreen validate "${FILE}"
+done
```

Comment:
[nit] Probably a bit early to add this line in, until we have integration with EVG. It's an early optimisation smell.